src/detail/fastrtps_dynamic_data.cpp

```diff
@@ -324,9 +515,13 @@ fastrtps__dynamic_data_serialize(
   }
   buffer->buffer_capacity = data_length;
 
-  auto payload = std::make_shared<eprosima::fastrtps::rtps::SerializedPayload_t>(
-    fastrtps__size_t_to_uint32_t(data_length));
-  bool success = m_type->serialize(data_impl->handle, payload.get());  // Serialize into payload
+  auto payload =
+    std::make_shared<eprosima::fastdds::rtps::SerializedPayload_t>(
+    fastdds__size_t_to_uint32_t(
+      data_length));
+  bool success = m_type->serialize(
+    &data_handle->ref_type, *payload,
+    eprosima::fastdds::dds::DataRepresentationId_t::XCDR_DATA_REPRESENTATION);                                                                    // Serialize into payload
```

Comment:
I know we aren't enforcing linting here, but I think we can shrink the whitespace here a lot:

```suggestion
    eprosima::fastdds::dds::DataRepresentationId_t::XCDR_DATA_REPRESENTATION);  // Serialize into payload
```